### PR TITLE
ollama: Fix self.spec reference

### PR DESCRIPTION
**Pull request by wspear**
@teaguesterling

### CODE REVIEW

**Review by teaguesterling — APPROVED** [review]
Waiting for CI to finish, but LGTM. (Discussed over Slack prior to PR).

**Comment by teaguesterling** [comment]
@tldahlgren would you mind also reviewing this one? It's somewhat related to #47654 but fixes an issue that @wspear noticed in the cuda building.

File: var/spack/repos/builtin/packages/ollama/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ollama/package.py:46]: Great catch!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
